src/api/Middleware/groups.js

```diff
@@ -0,0 +1,33 @@
+import {
+  getCustomGroupsOrdering,
+  updateCustomGroupOrdering
+} from '../Actions';
+import actionTypes from '../Actions/actionTypes';
+
+const getGuiGroupsOrdering = async (luaApi, callback) => {
+  const object = await luaApi.guiTreeOrdering();
+  const orderingMap = object[1];
+  if (!orderingMap) {
+    return;
+  }
+  // Use a callback since we need to wait for the value to be returned
```

Comment:
You can do this with a .catch .then statement

```suggestion
 const getGuiGroupsOrdering = async (luaApi, dispatch) => {
  await luaApi.guiTreeOrdering()
  .catch((e) => console.log(e))
  .then((data) => {
    const orderingMap = data[1];
    if (!orderingMap) {
      console.error("Error retrieving GUI tree ordering");
    }
    dispatch(updateCustomGroupOrdering(orderingMap));
  });
};

const groups = (store) => (next) => (action) => {
  const result = next(action);
  switch (action.type) {
    case actionTypes.initializeLuaApi:
      store.dispatch(getCustomGroupsOrdering(action.payload));
      break;
    case actionTypes.getCustomGroupsOrdering:
      getGuiGroupsOrdering(action.payload, store.dispatch);
      break;
    default:
      break;
  }
  return result;
};
export default groups;

```